src/cascadia/TerminalApp/CascadiaSettingsSerialization.cpp

```diff
@@ -605,19 +607,19 @@ void CascadiaSettings::_LayerOrCreateProfile(const Json::Value& profileJson)
         // If this JSON represents a dynamic profile, we _shouldn't_ create the
         // profile here. We only want to create profiles for profiles without a
         // `source`. Dynamic profiles _must_ be layered on an existing profile.
-        if (!Profile::IsDynamicProfileObject(profileJson))
+        if (!implementation::Profile::IsDynamicProfileObject(profileJson))
         {
-            Profile profile{};
+            auto profile = winrt::make_self<implementation::Profile>();
 
             // GH#2325: If we have a set of default profile settings, apply them here.
             // We _won't_ have these settings yet for defaults, dynamic profiles.
             if (_userDefaultProfileSettings)
             {
-                profile.LayerJson(_userDefaultProfileSettings);
+                profile->LayerJson(_userDefaultProfileSettings);
             }
 
-            profile.LayerJson(profileJson);
-            _profiles.emplace_back(profile);
+            profile->LayerJson(profileJson);
+            _profiles.emplace_back(profile.as<Profile>());

```

Comment:
```suggestion
            _profiles.emplace_back(*profile);
```

this should be sufficient.